src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php

```diff
@@ -250,6 +250,13 @@ public function handle(Request $request, $type = HttpKernelInterface::MASTER_REQ
      */
     public function terminate(Request $request, Response $response)
     {
+        // Do not call any listeners in case of a cache hit.
+        // This ensures identical behavior as if you had a separate
+        // reverse caching proxy such as Varnish and the like.
+        if (\in_array('fresh', $this->traces[$this->getTraceKey($request)] ?? [], true)) {
```

Comment:
is there a no more explict way to determine if we had a cache hit?

Reply:
Not really, no. I mean I can store the information in yet another variable but this would be the same as storing it in `$this->traces`. Imho it's pretty clean as this is internal to this very class and cannot be tampered with. Checking the header on the response would be a lot worse because the traces can have different formats.